lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -4,378 +4,95 @@
-    public function beginTransaction()
-    {
-        $this->connect('master');
-
-        return parent::beginTransaction();
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function commit()
-    {
-        $this->connect('master');
+        if ($connectionName === 'slave') {
+            $connectionName = 'replica';
 
-        return parent::commit();
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function rollBack()
-    {
-        $this->connect('master');
-
-        return parent::rollBack();
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function delete($tableName, array $identifier, array $types = [])
-    {
-        $this->connect('master');
-
-        return parent::delete($tableName, $identifier, $types);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function close()
-    {
-        unset($this->connections['master'], $this->connections['slave']);
-
-        parent::close();
-
-        $this->_conn       = null;
-        $this->connections = ['master' => null, 'slave' => null];
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function update($tableName, array $data, array $identifier, array $types = [])
-    {
-        $this->connect('master');
-
-        return parent::update($tableName, $data, $identifier, $types);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function insert($tableName, array $data, array $types = [])
-    {
-        $this->connect('master');
-
-        return parent::insert($tableName, $data, $types);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function exec($statement)
-    {
-        $this->connect('master');
-
-        return parent::exec($statement);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function createSavepoint($savepoint)
-    {
-        $this->connect('master');
-
-        parent::createSavepoint($savepoint);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function releaseSavepoint($savepoint)
-    {
-        $this->connect('master');
-
-        parent::releaseSavepoint($savepoint);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function rollbackSavepoint($savepoint)
-    {
-        $this->connect('master');
-
-        parent::rollbackSavepoint($savepoint);
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function query()
-    {
-        $this->connect('master');
-        assert($this->_conn instanceof DriverConnection);
-
-        $args = func_get_args();
-
-        $logger = $this->getConfiguration()->getSQLLogger();
-        if ($logger) {
-            $logger->startQuery($args[0]);
-        }
-
-        $statement = $this->_conn->query(...$args);
-
-        $statement->setFetchMode($this->defaultFetchMode);
-
-        if ($logger) {
-            $logger->stopQuery();
+            $this->deprecated('connect("slave")', 'ensureConnectedToReplica()');
         }
 
-        return $statement;
+        return parent::connect($connectionName);
```

Comment:
This is broken. When passing `slave` or `master` as connection name, you must call the alternative method instead of calling `parent::connect($connectionName);` with a renamed connection name, as you changed the parent class.

`parent::connect()` should be called only for remaining cases